exp/api/v1beta1/azuremanagedcluster_webhook.go

```diff
@@ -71,6 +71,26 @@ func (r *AzureManagedCluster) ValidateUpdate(oldRaw runtime.Object) error {
 				fmt.Sprintf("annotations with '%s' prefix are immutable", azure.CustomHeaderPrefix)))
 	}
 
+	if old.Spec.ControlPlaneEndpoint.Host != "" {
```

Comment:
Are there any cases where the controller would need to change either of these? Because webhook validations also apply to patches made by controllers, this essentially prevents the CAPZ controllers from changing these values too.

Reply:
I did some investigation and it seems that AKS does not have any notion of ever being able to change the apiserver endpoint.

Reply:
How would this change affect a scenario where the user has put in wrong/invalid/ill-formed `ControlPlaneEndpoint` ? Do we validate initial `ControlPlaneEndpoint` ?

Reply:
Good question @nawazkh!

In fact this is one of those "weird" fields where we defer to the AKS service itself to populate the value (and thus we trust that the AKS service will always populate the right value).

@CecileRobertMichon @nojnhuh is there any additional foo we should put in place here? It does feel a bit weird not having any create webhook validations for such an important field, but on the other hand this is an exceptional case where we're not actually able to implement what we want:

1. Never allow the user to ever populate this field!
2. Always trust the AKS service and accept whatever it provides!

Is there a better way we could be doing this, or is "don't enforce CREATE webhooks" the most appropriate given the state of AKS + capz?

Reply:
The real issue here is that controlPlaneEndpoint should be a Status field, not a Spec field, since it's meant for the CAPZ controller to communicate back info to CAPI, not to be user configurable.

I had a write-up on this in https://github.com/kubernetes-sigs/cluster-api/issues/3715 but unfortunately in never moved forward.

Reply:
That makes sense. My interpretation of that is that we're doing the best we can right now.

Reply:
Is the behavior currently that if the `ControlPlaneEndpoint` is changed, that will trigger a reconciliation where the controller will immediately override the provided value with AKS's value? If so, then to @nawazkh's point, it seems like this removes the possibility for the controller to automatically reset the value in case it does happen to change. e.g. If a user mistakenly sets this before the controller does, it seems like the only way to resolve this is to delete and recreate the control plane. Overall I think this check is valuable, but worth being mindful of that change.

One other idea: Is there a way to make this trigger a warning instead of a hard error? Then the webhook could say something to the effect of, "You're trying to change this value, but ultimately we're going to ignore that and set it to what AKS tells us it should be."

Reply:
If the *user* is responsible for changing from `""` --> `<some value>`, then that `<some value>` will become the durable one, and the automated AKS API-informed update that happens during cluster create reconciliation will be webhook-rejected with this change.

I'm certainly open to any comments on how we can improve this! It seemed to my initially naive understanding that the current situation, with no enforcement upon this rather critical configuration vector, left room for improvement.